Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -362,5 +750,396 @@ ReflectometryReductionOne2::convertToQ(MatrixWorkspace_sptr inputWS) {
+    outputY[outIdx] += inputCounts * fraction;
+    outputE[outIdx] += inputErr * fraction;
+  }
+}
+
+/**
+* Project an input pixel onto an arbitrary reference line at twoThetaR. The
+* projection is done along lines of constant Q, which emanate from theta0. The
+* top-left and bottom-right corners of the pixel are projected, resulting in an
+* output range in "virtual" lambda (lambdaV).
+*
+* @param lambda [in] :: the lambda coord of the centre of the pixel to project
+* @param twoTheta [in] :: the twoTheta coord of the centre of the pixel to
+*project
+* @param bLambda [in] :: the pixel size in lambda
+* @param bTwoTheta [in] :: the pixel size in twoTheta
+* @param detectors [in] :: spectrum indices of the detectors of interest
+* @param lambdaVMin [out] :: the projected range start
+* @param lambdaVMax [out] :: the projected range end
+*/
+void ReflectometryReductionOne2::getProjectedLambdaRange(
+    const double lambda, const double twoTheta, const double bLambda,
+    const double bTwoTheta, const std::vector<size_t> &detectors,
+    double &lambdaVMin, double &lambdaVMax) {
+
+  // Get the angle from twoThetaR to this detector
+  const double twoThetaRVal = twoThetaR(detectors);
+  // Get the distance from the pixel to twoThetaR
+  const double gamma = twoTheta - twoThetaRVal;
+  // Get the angle from the horizon to the reference angle
+  const double horizonThetaR = twoThetaRVal - theta0();
+
+  // Calculate the projected wavelength range
+  try {
+    const double lambdaTop = std::sin(horizonThetaR) *
+                             (lambda + bLambda / 2.0) /
+                             std::sin(horizonThetaR + gamma - bTwoTheta / 2.0);
+    const double lambdaBot = std::sin(horizonThetaR) *
+                             (lambda - bLambda / 2.0) /
+                             std::sin(horizonThetaR + gamma + bTwoTheta / 2.0);
+    lambdaVMin = std::min(lambdaTop, lambdaBot);
+    lambdaVMax = std::max(lambdaTop, lambdaBot);
+  } catch (std::exception &ex) {
+    std::stringstream errMsg;
+    errMsg << "Failed to project (lambda, twoTheta) = (" << lambda << ","
+           << twoTheta * 180.0 / M_PI << ") onto twoThetaR = " << twoThetaRVal
+           << ": " << ex.what();
+    throw std::runtime_error(errMsg.str());
+  }
+}
+
+/**
+Check whether the spectrum maps for the given workspaces are the same.
+
+@param ws1 : First workspace to compare
+@param ws2 : Second workspace to compare against
+@param severe: True to indicate that failure to verify should result in an
+exception. Otherwise a warning is generated.
+*/
+void ReflectometryReductionOne2::verifySpectrumMaps(
+    MatrixWorkspace_const_sptr ws1, MatrixWorkspace_const_sptr ws2,
+    const bool severe) {
+  auto map1 = ws1->getSpectrumToWorkspaceIndexMap();
+  auto map2 = ws2->getSpectrumToWorkspaceIndexMap();
+  if (map1 != map2) {
+    const std::string message =
+        "Spectrum maps between workspaces do NOT match up.";
+    if (severe) {
```

Comment:
This is passed through from the `StrictSpectrumChecking` property, which is true by default. I think you would set it to false if you want to normalise by a transmission workspace that has different spectrum numbers to the detector workspace.